.github/actions/tr-ultimate-test-harness/tr-ultimate-test-harness.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+trap 'echo "Error on line ${LINENO} of ${0}"; exit 1' ERR;
+set -o xtrace
+set -o errexit -o nounset -o pipefail
+docker-compose pull
+docker-compose build --parallel
+docker-compose up -d
+
+# Constants
+declare -r cookie_name=dev-ciab-cookie
+
+# Set TO_USER, TO_PASSWORD, and TO_URL environment variables and get atc-ready function
+source dev/atc.dev.sh
+
+until atc-ready; do
+	echo 'Waiting until Traffic Ops is ready to accept requests...'
+	sleep 3
+done
+echo 'Traffic Ops is ready to accept requests!'
+
+to-req() {
+	endpoint="$1"
+	shift
+	local curl_command=(curl --insecure --silent --cookie-jar "$cookie_name" --cookie "$cookie_name" "${TO_URL}/api/${API_VERSION}")
+	"${curl_command[@]}${endpoint}" "$@" | jq
+}
```

Comment:
If you use [the scripts provided by `toaccess`](https://traffic-control-cdn.readthedocs.io/en/latest/tools/toaccess.html#id1) (or the similar [`to_access.sh` script](https://github.com/apache/trafficcontrol/blob/f24c0b8257c8762e774c0154618294863a066a1a/infrastructure/cdn-in-a-box/traffic_ops/to-access.sh) for less dependencies) you won't need to implement this yourself. This will be the third time (of which I'm aware) that some function like this is written in our repo.

Reply:
Fourth https://github.com/apache/trafficcontrol/blob/741a09fba1b793b2c0c8a8771d55cd3d2e85dd0e/traffic_ops/install/bin/copy-pgdata.sh#L75

Reply:
:facepalm:

Reply:
Reusing `to-access.sh` from CDN in a Box in 5751315